lib/Db/Provider.php

```diff
@@ -56,14 +55,21 @@ class Provider extends Entity implements \JsonSerializable {
 	/** @var string */
 	protected $scope;
 
+	/**
+	 * @return string
+	 */
+	public function getScope(): string {
+		return $this->scope ?: ' ';
+	}
+
 	#[\ReturnTypeWillChange]
 	public function jsonSerialize() {
 		return [
 			'id' => $this->id,
 			'identifier' => $this->identifier,
 			'clientId' => $this->clientId,
 			'discoveryEndpoint' => $this->discoveryEndpoint,
-			'scope' => $this->scope,
+			'scope' => trim($this->scope),
```

Comment:
```suggestion
			'scope' => trim($this->getScope()),
```

Reply:
Could also move the trim inside the getter then ;)

Reply:
Hmm I think we can't. It seems when updating/inserting a row in the DB, the value that is put there is accessed via the getter. So if we trim there, we will insert/update with an empty string which is what we want to avoid.

Reply:
Right, fair enough then

Reply:
Also, why do you think we should use the getter in jsonSerialize? For empty scopes, the getter will replace it by a space and then we trim it anyway.